core/src/main/scala/filodb.core/store/ChunkSetInfo.scala

```diff
@@ -244,34 +250,36 @@ trait ChunkInfoIterator { base: ChunkInfoIterator =>
 
 object ChunkInfoIterator {
   val empty = new ChunkInfoIterator {
+    def close(): Unit = {}
     def hasNext: Boolean = false
     def nextInfo: ChunkSetInfo = ChunkSetInfo(0)
   }
-
-  def single(info: ChunkSetInfo): ChunkInfoIterator = new ChunkInfoIterator {
-    var count = 0
-    def hasNext: Boolean = count == 0
-    def nextInfo: ChunkSetInfo = { count += 1; info }
-  }
 }
 
 class ElementChunkInfoIterator(elIt: ElementIterator) extends ChunkInfoIterator {
+  def close(): Unit = elIt.close()
```

Comment:
I dont see a call for the iterator's close() method from anywhere except the tests. Did I miss it? I was expecting for it to be invoked from PartitionTimeRangeReader class where we iterate across the chunks.

Reply:
All the uses of the iterator that I could see always iterate to the end. Are there cases I missed?

Reply:
Added some counters.

Reply:
Added a releaseAllSharedLocks which runs when the query finishes. This covers all missed cases.